packages/wonder-blocks-announcer/src/__tests__/announce-message.test.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from "react";
+import {render, screen, waitFor} from "@testing-library/react";
+import Announcer, {REMOVAL_TIMEOUT_DELAY} from "../announcer";
+import {AnnounceMessageButton} from "./components/announce-message-button";
+import {announceMessage} from "../announce-message";
+
+jest.useFakeTimers();
+jest.spyOn(global, "setTimeout");
+
+describe("Announcer.announceMessage", () => {
+    afterEach(() => {
+        const announcer = Announcer.getInstance();
+        jest.advanceTimersByTime(REMOVAL_TIMEOUT_DELAY);
+        announcer.reset();
+    });
+
+    test("returns a targeted element IDREF", async () => {
+        // ARRANGE
+        const message1 = "One Fish Two Fish";
+
+        // ACT
+        const announcement1Id = await announceMessage({
+            message: message1,
+            initialTimeout: 0,
+            debounceThreshold: 0,
+        });
+        jest.advanceTimersByTime(500);
+
+        // ASSERT
+        expect(announcement1Id).toBe("wbARegion-polite1");
+    });
+
+    test("creates the live region elements when called", () => {
+        // ARRANGE
+        const message = "Ta-da!";
+        render(
+            <AnnounceMessageButton message={message} debounceThreshold={0} />,
+        );
+
+        // ACT: call function
+        const button = screen.getByRole("button");
+        button.click();
+
+        // ASSERT: expect live regions to exist
+        const wrapperElement = screen.getByTestId("wbAnnounce");
+        const regionElements = screen.queryAllByRole("log");
+        expect(wrapperElement).toBeInTheDocument();
+        expect(regionElements).toHaveLength(4);
+    });
+
+    test("appends to polite live regions by default", () => {
+        // ARRANGE
+        const message = "Ta-da, nicely!";
+        render(
+            <AnnounceMessageButton message={message} debounceThreshold={0} />,
+        );
+
+        // ACT: call function
+        const button = screen.getByRole("button");
+        button.click();
+
+        // ASSERT: expect live regions to exist
+        const politeRegion1 = screen.queryByTestId("wbARegion-polite0");
+        const politeRegion2 = screen.queryByTestId("wbARegion-polite1");
+        expect(politeRegion1).toHaveAttribute("aria-live", "polite");
+        expect(politeRegion1).toHaveAttribute("id", "wbARegion-polite0");
+        expect(politeRegion2).toHaveAttribute("aria-live", "polite");
+        expect(politeRegion2).toHaveAttribute("id", "wbARegion-polite1");
+    });
```

Comment:
Question: Can we make these `aria-atomic` so that they read the whole message instead of browsers/srs trimming similar starting text? Or is that handled in a different way?

Reply:
We can definitely try it! I wasn't noticing differences with the current implementation. But I will caveat that I'm also planning to experiment with a queue of messages. So `aria-atomic` might not make sense in that implementation!

Reply:
I made a note to look into this in a future PR, since I want to do some more cross-platform testing anyway!